In [2]:
import pandas as pd
import numpy as np
from sklearn.cluster import DBSCAN
import matplotlib.pyplot as plt

In [8]:
data = pd.read_csv(
    'https://archive.ics.uci.edu/ml/machine-learning-databases/00292/Wholesale%20customers%20data.csv')

eps_values = [0.5, 1, 1.5, 2, 2.5]
min_samples_values = [5, 10, 15, 20, 25]

results_list = []

for eps in eps_values:
    for min_samples in min_samples_values:
        dbscan = DBSCAN(eps=eps, min_samples=min_samples)
        clusters = dbscan.fit_predict(data)

        unique_clusters, counts_clusters = np.unique(
            clusters, return_counts=True)
        n_clusters = len(unique_clusters)
        cluster_sizes = ", ".join([str(count) for count in counts_clusters])

        results_list.append({
            'eps': eps,
            'min_samples': min_samples,
            'n_clusters': n_clusters,
            'cluster_sizes': cluster_sizes
        })

results = pd.DataFrame(results_list)
print(results)


    eps  min_samples  n_clusters cluster_sizes
0   0.5            5           1           440
1   0.5           10           1           440
2   0.5           15           1           440
3   0.5           20           1           440
4   0.5           25           1           440
5   1.0            5           1           440
6   1.0           10           1           440
7   1.0           15           1           440
8   1.0           20           1           440
9   1.0           25           1           440
10  1.5            5           1           440
11  1.5           10           1           440
12  1.5           15           1           440
13  1.5           20           1           440
14  1.5           25           1           440
15  2.0            5           1           440
16  2.0           10           1           440
17  2.0           15           1           440
18  2.0           20           1           440
19  2.0           25           1           440
20  2.5      

bu sonuçlara göre anlamlı bir kümeleme olmamıştır. Tüm parametre kombinasyonları için yalnızca bir küme oluşmuştur ve her kümenin boyutu 440'dır.

In [5]:
data = pd.read_csv(
    'https://archive.ics.uci.edu/ml/machine-learning-databases/00292/Wholesale%20customers%20data.csv')

eps_values = [0.5, 1, 1.5, 2, 2.5]
min_samples_values = [5, 10, 15, 20, 25]

results_list = []

for eps in eps_values:
    for min_samples in min_samples_values:
        dbscan = DBSCAN(eps=eps, min_samples=min_samples)
        clusters = dbscan.fit_predict(data)

        unique_clusters, counts_clusters = np.unique(
            clusters, return_counts=True)
        n_clusters = len(unique_clusters)
        cluster_sizes = ", ".join([str(count) for count in counts_clusters])

        results_list.append({
            'eps': eps,
            'min_samples': min_samples,
            'n_clusters': n_clusters,
            'cluster_sizes': cluster_sizes
        })

results = pd.DataFrame(results_list)
print(results)


    eps  min_samples  n_clusters cluster_sizes
0   0.5            5           1           440
1   0.5           10           1           440
2   0.5           15           1           440
3   0.5           20           1           440
4   0.5           25           1           440
5   1.0            5           1           440
6   1.0           10           1           440
7   1.0           15           1           440
8   1.0           20           1           440
9   1.0           25           1           440
10  1.5            5           1           440
11  1.5           10           1           440
12  1.5           15           1           440
13  1.5           20           1           440
14  1.5           25           1           440
15  2.0            5           1           440
16  2.0           10           1           440
17  2.0           15           1           440
18  2.0           20           1           440
19  2.0           25           1           440
20  2.5      

Bu sonuçlara göre de farklı epsilon ve min samples değerleri seçilmiş olmasına rağmen benzer sonuç çıkmıştırç kümeleme boyutu 440 olmuştur.